pkg/sql/pgwire/conn.go

```diff
@@ -313,7 +313,10 @@ Loop:
 		typ, n, err = c.readBuf.ReadTypedMsg(&c.rd)
 		c.metrics.BytesInCount.Inc(int64(n))
 		if err != nil {
-			break Loop
+			if err = c.stmtBuf.Push(ctx, sql.SendError{Err: err}); err != nil {
```

Comment:
My understanding is that this code previously would kill the connection if ReadTypedMsg returned an error, and this new code will instead return the error to the client and then resume. Is that correct? If so, I assume the "regain stream sync" thing below is the thing that will allow the connection to continue. If that's correct, I'd like to see a test that verifies the behavior that after the "msg too big" error, that same connection can still do other queries.

Reply:
Yes, we used to kill the connection completely. Postgres doesn't do that in this case, so I'm attempting to not do that either.

Good idea, I'll add that test.